STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+  - [Use truthy/falsy checks where possible](#use-truthyfalsy-checks-where-possible)
+- [White space](#white-space)
+  - [Indentation: Use 2 spaces, not tabs ✓](#indentation-use-2-spaces-not-tabs-)
+  - [Add one empty line before and after function definitions](#add-one-empty-line-before-and-after-function-definitions)
+- [Comments](#comments)
+  - [Document public API with TypeDoc comments](#document-public-api-with-typedoc-comments)
+  - [Use `@deprecated` tag for deprecation](#use-deprecated-tag-for-deprecation)
+  - [Don't reference types in parameters an type parameters](#dont-reference-types-in-parameters-an-type-parameters)
+  - [Use @returns if a function has a return value ✓](#use-returns-if-a-function-has-a-return-value-)
+- [Tests](#tests)
+  - [Use .spec in test file names](#use-spec-in-test-file-names)
+  - [Use it and describe notation](#use-it-and-describe-notation)
+  - [Put unit test files next to source files](#put-unit-test-files-next-to-source-files)
+- [Promises](#promises)
+  - [Use `async`/`await` rather than `.then`](#use-asyncawait-rather-than-then)
+- [Functions](#functions)
+  - [Use arrow functions for callbacks](#use-arrow-functions-for-callbacks)
+  - [Use function declarations to reference functions by name](#use-function-declarations-to-reference-functions-by-name)
+
+## Naming
+
+### Use kebap case for file names
+Use kebap case for all file names and directories, except for generated files and markdown files, that are not in the knowledge base.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use camel case */
+someDir/someClass.ts
+
+/* Don't use pascal case */
+SomeDir/SomeClass.ts
+
+/* Don't use snake case */
+some_dir/some_class.ts
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use kebap case */
+some-dir/some-class.ts
+```
+
+### Use camel case for variable names ✓
+This is common in most JavaScript/TypeScript projects.
+This rule applies to **all** variables, even static ones or reusable constants.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use uppercase snake case */
+const FOO_BAR = 'foo';
+
+/* Don't use snake case */
+const foo_bar = 'foo';
+
+/* Don't use different case for static variables */
+class FooBar {
+  static FOO_BAR;
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use camel case */
+const fooBar = 'foo';
+
+/* Use different case for static variables */
+class FooBar {
+  static fooBar;
+}
+```
+
+### Use pascal case for classes ✓
+Pascal case is common in most JavaScript/TypeScript projects for classes.
+
+✅ Examples of **correct** code:
+```ts
+/* Use pascal case */
+class FooBar {
+  ...
+}
+```
+
+### Use pascal case for interface names and don't prefix them
+Pascal case is common in most JavaScript/TypeScript projects for interfaces.
+Don't prefix interfaces with the hungarian notation 'I' (or other prefixes).
```

Comment:
Do you have an example where this would be the case? I think in most cases it makes sense to find a more general name for the interface or a more specific name for the class, but there are always exceptions to this rule. I'll add a sentence about this, let me know if you think there should be more.

Reply:
I was thinking about something like "BusinessPartner" as an interface containing all fields that describe a BuPa coming from an API and "BusinessPartner" the class that is created by the SDK to easily use the BusinessPartner instance.

I think general name for the interface and specific for class is fine but maybe an example helps so people don't replace the prefix with a bad suffix or something else.

In the example `interface BusinessPartnerResponse` and `class BusinessPartner` would actually go against the guidance but might fit well.

Reply:
Thanks for the clarification. Done.